### PR TITLE
ISPN-14636 LSET command

### PR DESCRIPTION
**Pull request by karesti**
https://issues.redhat.com/browse/ISPN-14636

### CODE REVIEW

File: multimap/src/test/java/org/infinispan/multimap/impl/EmbeddedMultimapListCacheTest.java [hunk 1]
```diff
@@ -302,6 +304,59 @@ public void testPollLast() {
...
+      assertThat(await(listCache.set(NAMES_KEY, -1, JULIEN))).isTrue();
+      assertThat(await(listCache.index(NAMES_KEY, -1))).isEqualTo(JULIEN);
+
+      // middle
+      assertThat(await(listCache.index(NAMES_KEY, 1))).isEqualTo(ELAIA);
+      assertThat(await(listCache.set(NAMES_KEY, 1, OIHANA))).isTrue();
+      assertThat(await(listCache.index(NAMES_KEY, 1))).isEqualTo(OIHANA);
+
+      assertThat(await(listCache.index(NAMES_KEY, -2))).isEqualTo(KOLDO);
+      assertThat(await(listCache.set(NAMES_KEY, 2, ELAIA))).isTrue();
```
Comment (jabolina) [inline multimap/src/test/java/org/infinispan/multimap/impl/EmbeddedMultimapListCacheTest.java:329]: `-2`? Just for completeness, we tested only the tail case.

File: server/resp/src/main/java/org/infinispan/server/resp/commands/list/LSET.java [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+         if (t != null) {
+           return handleException(handler, t);
+         }
+
+         if (!result) {
+            RespErrorUtil.noSuchKey(handler.allocatorToUse());
+            return handler.myStage();
+         }
+
+         return handler.stageToReturn(CompletableFuture.completedFuture(null), ctx, Consumers.OK_BICONSUMER);
```
Comment (jabolina) [inline server/resp/src/main/java/org/infinispan/server/resp/commands/list/LSET.java:64]: `CompletableFutures.completedNull()` instead.

**Review by jabolina — APPROVED** [review]
Looks good, just a couple of small things.

**Comment by karesti** [comment]
@jabolina reported changes

**Review by jabolina — APPROVED** [review]

File: multimap/src/main/java/org/infinispan/multimap/impl/ListBucket.java [hunk 3]
```diff
@@ -87,13 +87,62 @@ public String toString() {
...
+
+      // set head
+      if (index == 0) {
+         values.pollFirst();
+         values.offerFirst(value);
+         return new ListBucket<>(values);
+      }
+
+      // set tail
+      if (index == -1) {
```
Comment (wburns) [inline multimap/src/main/java/org/infinispan/multimap/impl/ListBucket.java:112]:
```suggestion
      if (index == -1 || index == values.size -1) {
```
Just a small optimization if they ended up picking the last node.

**Review by jabolina — APPROVED** [review]

**Comment by jabolina** [comment]
Integrated into main, thanks @karesti!

**Pull request merged**